cli/azd/test/functional/cli_test.go

```diff
@@ -68,17 +69,28 @@ func Test_CLI_Version_PrintsVersion(t *testing.T) {
 	defer cancel()
 
 	cli := azdcli.NewCLI(t)
-	out, err := cli.RunCommand(ctx, "version")
+	textOutput, err := cli.RunCommand(ctx, "version")
 	require.NoError(t, err)
 
-	rn := os.Getenv("GITHUB_RUN_NUMBER")
-	if rn != "" {
-		version := os.Getenv("CLI_VERSION")
-		require.NotEmpty(t, version)
-		require.Contains(t, out, version)
-	} else {
-		require.Contains(t, out, fmt.Sprintf("azd version %s", internal.Version))
+	jsonOutput, err := cli.RunCommand(ctx, "version", "--output", "json")
+	require.NoError(t, err)
+
+	versionJson := &internal.VersionSpec{}
+	err = json.Unmarshal([]byte(jsonOutput), versionJson)
```

Comment:
I wonder if we can use the `versionAction.Run()` and create a unit test. 
I would like us to start deprecating the `cli_test.go` and migrating to test the actions directly.
What do you think?

Reply:
@vhvb1989 It depends on whether we want an end-to-end test versus a unit test.

The end-to-end test is verifying that `azd version` is working as a whole, including that it was built with a version number that was instructed from the CI. The action unit test would be verifying the logic in `Run` which isn't what I want in this case..